### PR TITLE
Feature/m0228 j 473 update kifaru doc

### PR DESCRIPTION
**Pull request by bguedas**

### CODE REVIEW

File: ApplicationDeveloperGuide/js/java.rst [hunk 1]
```diff
@@ -37,7 +37,38 @@ Calling Java from JavaScript
 
 The MicroEJ engine allows to expose Java objects or methods to the JavaScript code by using the engine API and creating the adequate JavaScript object.
 
-For example, here is the code to create a JavaScript function named `javaPrint` in the global scope:
+Java objects can be exposed to JavaScript using the ``JavaImport`` mechanism:
```
Comment (tdelhomenie-microej) [inline ApplicationDeveloperGuide/js/java.rst:40]:
```suggestion
Java objects can be exposed to JavaScript using the ``JavaImport`` mechanism. 
```

File: ApplicationDeveloperGuide/js/java.rst [hunk 2]
```diff
@@ -37,7 +37,38 @@ Calling Java from JavaScript
 
 The MicroEJ engine allows to expose Java objects or methods to the JavaScript code by using the engine API and creating the adequate JavaScript object.
 
-For example, here is the code to create a JavaScript function named `javaPrint` in the global scope:
+Java objects can be exposed to JavaScript using the ``JavaImport`` mechanism: 
+it takes a Java fully qualified name as argument and returns an objects that
```
Comment (tdelhomenie-microej) [inline ApplicationDeveloperGuide/js/java.rst:41]:
```suggestion
It takes a Java fully qualified name as argument and returns an object that 
```

File: ApplicationDeveloperGuide/js/java.rst [hunk 3]
```diff
@@ -37,7 +37,38 @@ Calling Java from JavaScript
 
 The MicroEJ engine allows to expose Java objects or methods to the JavaScript code by using the engine API and creating the adequate JavaScript object.
 
-For example, here is the code to create a JavaScript function named `javaPrint` in the global scope:
+Java objects can be exposed to JavaScript using the ``JavaImport`` mechanism: 
+it takes a Java fully qualified name as argument and returns an objects that 
+gives access to the constructors, static methoods and static fields.
```
Comment (tdelhomenie-microej) [inline ApplicationDeveloperGuide/js/java.rst:42]:
```suggestion
gives access to the constructors, static methods and static fields.
```

File: ApplicationDeveloperGuide/js/java.rst [hunk 4]
```diff
@@ -37,7 +37,38 @@ Calling Java from JavaScript
...
+
+For instance, the following code imports ``java.lang.System`` and prints a 
+string calling ``System.out.println()``:
+
+.. code-block:: javascript
+
+	var System = JavaImport("java.lang.System")
+	System.out.println("foo");
+
+Here we instantiate a Java ``File`` object and checks that it exists:
```
Comment (tdelhomenie-microej) [inline ApplicationDeveloperGuide/js/java.rst:52]:
```suggestion
Here we instantiate a Java ``File`` object and check that it exists:
```

File: ApplicationDeveloperGuide/js/java.rst [hunk 5]
```diff
@@ -37,7 +37,38 @@ Calling Java from JavaScript
...
+		print("myFile.txt exists")
+	} else {
+		print("myFile.txt does not exist")
+	}
+
+.. warning::
+
+     You cannot instantiate an anonymous class from an interface or an abstract class with the ``new`` keyword and ``JavaImport``. Nevertheless, you can still access to static fields and methods.   
+
+We can also implement JavaScript functions in Java by adding their
```
Comment (tdelhomenie-microej) [inline ApplicationDeveloperGuide/js/java.rst:69]: Shoudn't we split in several sections, just to distinguish clearly the different ways to call Java from JavaScript ?

File: ApplicationDeveloperGuide/js/java.rst (hunk 3 above)
Comment (tdelhomenie-microej) [inline ApplicationDeveloperGuide/js/java.rst:42]: Maybe we should add a sentence to clearly state which Java classes can be imported (classes of the application and from the application dependencies).

File: ApplicationDeveloperGuide/js/java.rst [hunk 6]
```diff
@@ -123,7 +154,117 @@ it can be called from a Java piece of code with:
...
     JsObjectFunction functionObject = (JsObjectFunction) JsRuntime.JS_GLOBAL_OBJECT.get("sum");
-    JsRuntime.ENGINE.addJob(new Job(functionObject, JsRuntime.JS_GLOBAL_OBJECT, new Integer(5), new Integer(3)));
+    JsRuntime.ENGINE.addJob(functionObject, JsRuntime.JS_GLOBAL_OBJECT, new Integer(5), new Integer(3));
 
 The first line gets the JavaScript function from the global scope.
-The second line adds a job in the JavaScript engine queue to execute the function, in the global scope, with the arguments ``5`` and ``3``.
\ No newline at end of file
+The second line adds a job in the JavaScript engine queue to execute the function, in the global scope, with the arguments ``5`` and ``3``.
+
+Passing Values Between JavaScript and Java
```
Comment (tdelhomenie-microej) [inline ApplicationDeveloperGuide/js/java.rst:162]:
```suggestion
Passing values between JavaScript and Java
```

File: ApplicationDeveloperGuide/js/java.rst [hunk 7]
```diff
@@ -123,7 +154,117 @@ it can be called from a Java piece of code with:
...
+------------------------------------------
+
+JavaScript base types are represented by Java objects and not Java base types. 
+The following table show the correspondance between types in both languages:
```
Comment (tdelhomenie-microej) [inline ApplicationDeveloperGuide/js/java.rst:166]:
```suggestion
The following table shows the mapping between types in both languages: 
```

File: ApplicationDeveloperGuide/js/java.rst [hunk 8]
```diff
@@ -123,7 +154,117 @@ it can be called from a Java piece of code with:
...
+         */
+        public synchronized void returnValue(@Nullable T value) {
+            this.value = value;
+            this.returned = true;
+            notify();
+        }
+    }
+
+We can now pass the callback to the job. The Java code will wait on the 
+``callback.getValue()`` untill the result is ready.
```
Comment (tdelhomenie-microej) [inline ApplicationDeveloperGuide/js/java.rst:262]:
```suggestion
``callback.getValue()`` until the result is ready.
```

**Review by tdelhomenie-microej — CHANGES_REQUESTED** [review]

File: ApplicationDeveloperGuide/js/java.rst (hunk 1 above)
Comment (bguedas) [reply ApplicationDeveloperGuide/js/java.rst]: fixed

File: ApplicationDeveloperGuide/js/java.rst (hunk 2 above)
Comment (bguedas) [reply ApplicationDeveloperGuide/js/java.rst]: fixed

File: ApplicationDeveloperGuide/js/java.rst (hunk 5 above)
Comment (bguedas) [reply ApplicationDeveloperGuide/js/java.rst:77]: Done

File: ApplicationDeveloperGuide/js/java.rst (hunk 3 above)
Comment (bguedas) [reply ApplicationDeveloperGuide/js/java.rst]: Done

File: ApplicationDeveloperGuide/js/java.rst (hunk 7 above)
Comment (bguedas) [reply ApplicationDeveloperGuide/js/java.rst]: fixed

File: ApplicationDeveloperGuide/js/java.rst (hunk 8 above)
Comment (bguedas) [reply ApplicationDeveloperGuide/js/java.rst]: fixed

File: ApplicationDeveloperGuide/js/java.rst (hunk 6 above)
Comment (bguedas) [reply ApplicationDeveloperGuide/js/java.rst:170]: Titles are capitalized, why not these two words?

File: ApplicationDeveloperGuide/js/java.rst (hunk 4 above)
Comment (bguedas) [reply ApplicationDeveloperGuide/js/java.rst]: fixed

File: ApplicationDeveloperGuide/js/java.rst (hunk 6 above)
Comment (tdelhomenie-microej) [reply ApplicationDeveloperGuide/js/java.rst:170]: I don't know what is the rule exactly, but it looks weird to have words like "between" capitalized.

**Pull request merged**